crate/utils/src/crypto/hybrid_encryption/decryption.rs

```diff
@@ -59,40 +54,38 @@ impl DecryptionSystem for HybridDecryptionSystem {
         // Note: All conversions below will go once we move to full openssl
         let id = self.private_key.id();
         let plaintext = match id {
-            Id::EC => decrypt_with_nist_curve(&self.private_key, ciphertext)?,
-            Id::ED25519 => {
-                debug!("decrypt: match CURVEED25519");
-                let raw_bytes = self.private_key.raw_private_key()?;
-                let private_key_bytes: [u8; CURVE_25519_SECRET_LENGTH] = raw_bytes.try_into()?;
-                let private_key = Ed25519PrivateKey::try_from_bytes(private_key_bytes)?;
-                let private_key = X25519PrivateKey::from_ed25519_private_key(&private_key);
-                Zeroizing::new(EciesSalsaSealBox::decrypt(&private_key, ciphertext, None)?)
-            }
-            Id::X25519 => {
-                trace!("encrypt: X25519");
-                // The raw public key happens to be the (compressed) value of the Montgomery point
-                let raw_bytes = self.private_key.raw_private_key()?;
-                let private_key_bytes: [u8; CURVE_25519_SECRET_LENGTH] = raw_bytes.try_into()?;
-                let private_key = X25519PrivateKey::try_from_bytes(private_key_bytes)?;
-                Zeroizing::new(EciesSalsaSealBox::decrypt(&private_key, ciphertext, None)?)
-            }
+            Id::EC => Zeroizing::new(ecies_decrypt(&self.private_key, ciphertext)?),
             Id::RSA => {
                 if self.key_unwrapping {
                     Zeroizing::from(ckm_rsa_aes_key_unwrap(
                         self.private_key.clone(),
                         ciphertext,
                     )?)
                 } else {
-                    trace!("decrypt: RSA");
                     Zeroizing::from(rsa_oaep_aes_gcm_decrypt(
                         self.private_key.clone(),
                         ciphertext,
                         request.authenticated_encryption_additional_data.as_deref(),
                     )?)
                 }
             }
+            #[cfg(not(feature = "fips"))]
+            Id::ED25519 => {
+                let raw_bytes = self.private_key.raw_private_key()?;
+                let private_key_bytes: [u8; ED25519_SECRET_LENGTH] = raw_bytes.try_into()?;
+                let private_key = Ed25519PrivateKey::try_from_bytes(private_key_bytes)?;
+                let private_key = X25519PrivateKey::from_ed25519_private_key(&private_key);
+                Zeroizing::new(EciesSalsaSealBox::decrypt(&private_key, ciphertext, None)?)
+            }
+            #[cfg(not(feature = "fips"))]
+            Id::X25519 => {
+                // The raw public key happens to be the (compressed) value of the Montgomery point
```

Comment:
I'm confused as of what you point out to be an error ?

Reply:
There is no link with the code since you are managing private keys (this is the encryption function). My guess is that you copied/pasted the decryption function, modified it but forgot to remove this comment.

Reply:
Indeed I c/p this hunk of code to put the non-fips matching braces under others.
Do you think this comment should be removed ?

Reply:
It is already present where it may be needed : where the code is actually managing public keys. Here, except if I am missing something, we don't care about how public keys are representing because the code performs private key representation conversions.

So yes, I would remove it.